mobile-app/app/screens/WalletNavigator/screens/CreateWallet/VerifyMnemonicWalletV2.tsx

```diff
@@ -0,0 +1,217 @@
+          style={tailwind('px-4')}
+        />
+
+        <ThemedTextV2
+          style={tailwind('text-base mt-7 text-center font-normal-v2')}
+        >
+          {translate('screens/VerifyMnemonicWallet', 'Verify the written recovery words.')}
+        </ThemedTextV2>
+      </View>
+
+      {randomWords.map((n, index) => (
+        <RecoveryWordRow
+          index={n.index}
+          key={index}
+          lineNumber={index}
+          onWordSelect={(word) => {
+            selectedWords[n.index] = word
+            setSelectedWords([...selectedWords])
+            setValid(!selectedWords.some((w) => w === ''))
+          }}
+          words={n.words}
+        />
+      ))}
+
+      <ThemedTextV2
+        style={tailwind('text-base mt-7 text-center font-normal-v2')}
+      >
+        {translate('screens/VerifyMnemonicWallet', 'All questions must be answered correctly.')}
+      </ThemedTextV2>
+
+      <ButtonV2
+        styleProps='mt-5 mx-7'
+        delayLongPress={1000}
+        disabled={!isValid}
+        onLongPress={debugBypass}
+        onPress={onVerify}
+        testID='verify_words_button'
+        label={translate('screens/VerifyMnemonicWallet', 'Verify words')}
+      />
+    </ThemedScrollViewV2>
+  )
+}
+
+interface RecoveryWordItem {
+  index: number
+  words: string[]
+  onWordSelect: (word: string) => void
+  lineNumber: number
+}
+
+function RecoveryWordRow ({ index, words, onWordSelect, lineNumber }: RecoveryWordItem): JSX.Element {
+  const [selectedWord, setSelectedWord] = useState<string>()
+  return (
+    <View style={tailwind('mb-5')}>
+      <View style={tailwind('flex-row')}>
+        <ThemedTextV2
+          light={tailwind('text-mono-light-v2-400')}
+          dark={tailwind('text-mono-dark-v2-400')}
+          style={tailwind('text-xs font-normal-v2')}
+          testID={`line_${lineNumber}`}
+        >
+          {numberOrdinal(index + 1)}
+        </ThemedTextV2>
+        <ThemedTextV2
+          light={tailwind('text-mono-light-v2-400')}
+          dark={tailwind('text-mono-dark-v2-400')}
+          style={tailwind('text-xs ml-1 font-normal-v2')}
+        >
+          {translate('screens/VerifyMnemonicWallet', 'word')}
+        </ThemedTextV2>
+      </View>
+
+      <ThemedViewV2
+        light={tailwind('bg-mono-light-v2-00')}
+        dark={tailwind('bg-mono-dark-v2-00')}
+        style={tailwind('flex flex-row mt-2 justify-between items-center w-full rounded-3xl')}
+        testID={`recovery_word_row_${index}`}
+      >
+        {
+          words.map((w, i) => (
+            <ThemedTouchableOpacityV2
+              key={`${w}_${i}`}
+              onPress={() => {
+                setSelectedWord(w)
+                onWordSelect(w)
+              }}
+              dark={tailwind({ 'bg-mono-dark-v2-900': selectedWord === w })}
+              light={tailwind({ 'bg-mono-light-v2-900': selectedWord === w })}
+              style={tailwind('rounded py-3 w-1/4 rounded-3xl')}
```

Comment:
let's add horizontal padding too, just in case when user's viewport is small and we have long words
<img width="360" alt="image" src="https://user-images.githubusercontent.com/40191153/176082503-fb48fbae-adcd-4f91-b89e-ef213951b544.png">

Reply:
Added 4px x-axis padding